### PR TITLE
clear restore map interface for file and git

### PR DESCRIPTION
**Pull request by OsmondJiang**
1. split file and git into two map, because they are using totally different context
2. clear the fallback interface 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4923)

### CODE REVIEW

File: src/docfx/build/Build.cs [hunk 1]
```diff
@@ -36,7 +36,7 @@ public static async Task Run(string docsetPath, CommandLineOptions options, Erro
             string locale,
             CommandLineOptions options,
             ErrorLog errorLog,
-            RestoreMap restoreMap,
+            RestoreGitMap restoreMap,
```
Comment (live1206) [inline src/docfx/build/Build.cs]: change the parameter to `restoreGitMap` as well?

File: src/docfx/build/xref/ExternalXrefMapLoader.cs [hunk 2]
```diff
@@ -30,8 +30,7 @@ internal class ExternalXrefMapLoader
                 }
                 else
                 {
-                    // TODO: It is very easy to forget passing fallbackDocsetPath, the RestoreMap interface needs improvement
-                    var filePath = RestoreMap.GetRestoredFilePath(docset.DocsetPath, url, docset.FallbackDocset?.DocsetPath);
+                    var filePath = RestoreFileMap.GetRestoredFilePath(docset, url);
```
Comment (yufeih) [inline src/docfx/build/xref/ExternalXrefMapLoader.cs]: It is possible to abstract away all the unrelated concept, so the caller only needs to pass in an `url`?

```suggestion
                    var filePath = xxx.GetFileRestorePath(url);
```

File: src/docfx/build/Build.cs (hunk 1 above)
Comment (OsmondJiang) [reply src/docfx/build/Build.cs]: sounds good.

File: src/docfx/publish/PublishModelBuilder.cs [hunk 3]
```diff
@@ -18,8 +18,6 @@ internal class PublishModelBuilder
 
         public void MarkError(Document file)
         {
-            // TODO: If Error has a Document identifier, we can retrieve files with errors from
-            //       error log without explicitly call PublishModelBuilder.MarkError
             _filesWithErrors.Add(file);
```
Comment (yufeih) [inline src/docfx/publish/PublishModelBuilder.cs]: Leave this TODO: this is about decouple `PublishModelBuilder` from `Document` and use `FilePath` as the key to here

cc @928PJY

Comment (OsmondJiang) [reply src/docfx/publish/PublishModelBuilder.cs]: > Change to hide fallback path as we discussed

I have hidden the fallback path

**Pull request merged**